### DISCUSSION TITLE
How to start Wezterm directly into my WSL Ubuntu Setup

### DISCUSSION THREAD

**Discussion by Trid-collab** [wezterm/wezterm#3346] [Q&A] [answered]
I am new to Wezterm. Presently when I open Wezterm, it opens into the default Powershell Terminal which I believe is being picked up from the existing Windows Terminal setup I have. I then do a right click and select the second option listed `New Tab (domain WSL:Ubuntu - Ubuntu)` I want to open the Wezterm directly into this Ubuntu option. How can I configure this.

**Comment by wez** [comment, marked as answer]
Take a look at: https://wezfurlong.org/wezterm/config/lua/config/default_domain.html

**Reply by jesperalmstrom** [reply]
This did not work for me. What am I doing wrong? This is my lua config: 
```lua
-- Pull in the wezterm API
local wezterm = require 'wezterm'

-- This table will hold the configuration.
local config = {}
config.wsl_domains = {
  {
    -- The name of this specific domain.  Must be unique amonst all types
    -- of domain in the configuration file.
    name = 'WSL:Ubuntu',

    -- The name of the distribution.  This identifies the WSL distribution.
    -- It must match a valid distribution from your `wsl -l -v` output in
    -- order for the domain to be useful.
    distribution = 'Ubuntu',
  },
}
config.default_domain = 'WSL:Ubuntu'

-- In newer versions of wezterm, use the config_builder which will
-- help provide clearer error messages
if wezterm.config_builder then
  config = wezterm.config_builder()
end

config.font = wezterm.font 'MesloLGS Nerd Font Mono'

-- For example, changing the color scheme:
config.color_scheme = 'AdventureTime'

-- and finally, return the configuration to wezterm
return config
```
Still get the `cmd.exe` shell

**Reply by nekowinston** [reply]
@jesperalmstrom  You're overwriting config after previously setting it:
```
local wezterm = require 'wezterm'

local config = {}
if wezterm.config_builder then
  config = wezterm.config_builder()
end

-- change config now
config.default_domain = 'WSL:Ubuntu'
```
See it in the REPL:
![2024-03-14-181608@2x](https://github.com/wez/wezterm/assets/79978224/8b49df57-86f4-46cc-a72c-ebbf4a6d5d2d)

**Reply by jesperalmstrom** [reply]
Thanks, that helped a lot. Now I am able to ditch Alacritty :-)

**Reply by internetisaiah** [reply]
**Note:** If you're managing your `.wezterm.lua` from WSL and copying or syncing it up to your Windows user directory, you may run into this error even after setting `default_domain`:

```bash
⚠️  Process "wsl.exe ~" in domain "local" didn't exit cleanly
Exited with code 1.
This message is shown because exit_behavior="CloseOnCleanExit"
```

The `default_domain` approach works when WezTerm is configured directly on Windows, but when the config originates from WSL, WezTerm may not have `wsl.exe` on its PATH. To resolve this, I ended up using the full path to `wsl.exe` via `default_prog` instead:

```lua
config.default_prog = { 'C:\\Windows\\System32\\wsl.exe', '--distribution', 'Ubuntu', '--cd', '~' }
```

Just wanted to share in case other users are experiencing a similar issue. Feel free to follow up if you believe there's a way to improve this method 👍🏽
